addons/web/models/models.py

```diff
@@ -298,6 +292,320 @@ def web_resequence(self, specification: dict[str, dict], field_name: str = 'sequ
             record.write({field_name: i})
         return self.web_read(specification)
 
+    @api.model
+    @api.readonly
+    def web_read_group_unity(
+        self,
+        domain: DomainType,
+        groupby: list[str],
+        aggregates: list[str],
+        limit: int | None = None,
+        offset: int = 0,
+        *,
+        # Forced order from client or by the view in form of [{name: <field_name>, asc: bool}, ...]
+        forced_order: list[dict[str, bool | str]] | None = None,
+
+        # If we need to unfold group by default (true => kanban), if false still care about __fold
+        auto_unfold: bool = False,
+
+        # For reloading a specific group opening
+        # opening_info = [{
+        #      value: raw_value_groupby,
+        #      folded: True or False,
+        #      [offset: int,]
+        #      [limit: int,]
+        #      [extra_domain: extra_domain,] when click on a read_progress color
+        #      [groups: <opening_info>,]
+        # }]
+        opening_info: list[dict] | None = None,
+
+        # Arguments to read record inside the unfolded groups
+        unfold_read_specification: dict[str, dict] | None = None,
+        unfold_read_default_limit: int | None = 80,  # Limit of record by unfolded group by default
+
+        # To read extra value from the many2one groupby argument (only for list view for <groupby> leafs)
+        # {<groupby_spec>: <read_specification>}
+        groupby_read_specification: dict[str, dict] | None = None,
+    ) -> dict[str, int | list]:
+        """
+        :return: dict such as
+          ::
+
+            <result> = {
+                'groups': <groups>,
+                'length': int,
+            }
+            <groups> = <formatted_read_group(domain, [groupby[i]], ...)> but add recursive
+            __groups: <result> or __records: <web_read(unfold_read_specification)> to each group
+
+        """
+        assert groupby and isinstance(groupby, (list, tuple))
+        if '__count' not in aggregates:  # Used for computing length of sublevel groups
+            aggregates.append('__count')
+
+        # First level of grouping
+        dict_order = {
+            order_spec['name']: order_spec['asc'] for order_spec in forced_order
+        } if forced_order else {}
+        first_groupby = [groupby[0]]
```

Comment:
Add here `domain = Domain(domain)` so it does not have to be done in each function. You could even run basic optimization.